scripts/archive-dataset-in-unused-pipelines.py

```diff
@@ -0,0 +1,167 @@
+    for activity in activities:
+        datasets.update(extract_datasets_from_activity(activity))
+    return datasets
+
+def get_datasets_from_unused_pipelines(pipeline_dir, unused_pipelines):
+    all_datasets = set()
+    for pipeline_name in unused_pipelines:
+        file_path = os.path.join(pipeline_dir, f"{pipeline_name}.json")
+        if os.path.exists(file_path):
+            datasets = extract_datasets_from_pipeline_file(file_path)
+            all_datasets.update(datasets)
+    return all_datasets
+
+def update_dataset_paths_in_place(unused_datasets, dataset_dir, dataset_to_pipelines, unused_pipelines):
+    print(f"\n✏️ Updating dataset JSONs to use folder 'Archive/Unused' under Synapse workspace...\n")
+
+    for dataset in unused_datasets:
+        file_path = os.path.join(dataset_dir, f"{dataset}.json")
+        if not os.path.exists(file_path):
+            print(f" File not found: {file_path}")
+            continue
+
+        # Determine if any associated pipeline is still active
+        pipelines = dataset_to_pipelines.get(dataset, set())
+        is_used_in_active_pipeline = any(p not in unused_pipelines for p in pipelines)
+
+        # Only archive if ALL pipelines are unused
+        should_archive = not is_used_in_active_pipeline
+
+        with open(file_path, 'r', encoding='utf-8') as f:
+            try:
+                content = json.load(f)
+            except json.JSONDecodeError:
+                print(f"Invalid JSON in: {file_path}")
+                continue
+
+        changed = False
+        folder_section = content.setdefault("properties", {}).setdefault("folder", {})
+
+        if should_archive:
+            if folder_section.get("name") != "Archive/Unused":
+                folder_section["name"] = "Archive/Unused"
+                changed = True
+        else:
+            if folder_section.get("name") == "Archive/Unused":
+                del folder_section["name"]
+                changed = True
+
+        # Check and update folderPath expression
+        location = content["properties"].get("typeProperties", {}).get("location", {})
+        folder_path = location.get("folderPath")
+        if isinstance(folder_path, dict) and "value" in folder_path:
+            original_expr = folder_path["value"]
+            is_archived = original_expr.strip().startswith("@concat('archive/")
+            if should_archive and not is_archived:
+                folder_path["value"] = f"@concat('archive/', {original_expr})"
+                changed = True
+            elif not should_archive and is_archived:
+                # Strip the archive wrapper if present
+                inner_expr = original_expr.strip()[len("@concat('archive/', "):-1]
+                folder_path["value"] = inner_expr
+                changed = True
+
+        if changed:
+            with open(file_path, 'w', encoding='utf-8') as f:
+                json.dump(content, f, indent=4)
```

Comment:
This should be like`json.dump(content, f, indent="\t", ensure_ascii=False)` to ensure a clean write of the json. Currently if you inspect the git diff it shows loads of things have changed even though you're just changing 1 line of the json